### PR TITLE
Make propagation happen on AsyncResponse using CFX JAX-RS

### PR DESCRIPTION
**Pull request by amarziali**
# What Does This Do

When async response is suspended, on CXF jax-rs, the subsequent invoke methods (response resume, exception mappers) are not propagating at least the root servlet span. 
This PR is fixing this

# Motivation

# Additional Notes

Jira ticket: [APMS-10975]

[APMS-10975]: https://datadoghq.atlassian.net/browse/APMS-10975?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-java-agent/instrumentation/cxf-2.1/src/main/java/datadog/trace/instrumentation/cxf/InvokerInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  public static class PropagateSpanAdvice {
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static AgentScope beforeInvoke(@Advice.Argument(0) final Exchange exchange) {
+      if (AgentTracer.activeSpan() != null || exchange == null || exchange.getInMessage() == null) {
+        return null;
+      }
+      final Object span =
+          ServletHelper.getServletRequestAttribute(
+              exchange.getInMessage().get("HTTP.REQUEST"), HttpServerDecorator.DD_SPAN_ATTRIBUTE);
+      System.err.println("GOOOOT SPAN " + span);
```
Comment (PerfectSlayer) [inline dd-java-agent/instrumentation/cxf-2.1/src/main/java/datadog/trace/instrumentation/cxf/InvokerInstrumentation.java:64]: Can you remove `println` message? 😇

File: dd-java-agent/instrumentation/cxf-2.1/src/main/java/datadog/trace/instrumentation/cxf/InvokerInstrumentation.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  public void adviceTransformations(AdviceTransformation transformation) {
+    transformation.applyAdvice(
+        ElementMatchers.isMethod().and(NameMatchers.named("invoke")),
+        getClass().getName() + "$PropagateSpanAdvice");
+  }
+
+  public static class PropagateSpanAdvice {
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static AgentScope beforeInvoke(@Advice.Argument(0) final Exchange exchange) {
+      if (AgentTracer.activeSpan() != null || exchange == null || exchange.getInMessage() == null) {
```
Comment (PerfectSlayer) [inline dd-java-agent/instrumentation/cxf-2.1/src/main/java/datadog/trace/instrumentation/cxf/InvokerInstrumentation.java:58]: What about making inexpensive tests first? (like `exchange == null`)

**Review by PerfectSlayer — COMMENTED** [review]
Hey 👋 
Just some minor comments

File: dd-java-agent/instrumentation/cxf-2.1/src/main/java/datadog/trace/instrumentation/cxf/InvokerInstrumentation.java (hunk 1 above)
Comment (amarziali) [reply dd-java-agent/instrumentation/cxf-2.1/src/main/java/datadog/trace/instrumentation/cxf/InvokerInstrumentation.java:64]: oups thanks I forgot. Hopefully the debug message is polite :)

File: dd-java-agent/instrumentation/cxf-2.1/src/main/java/datadog/trace/instrumentation/cxf/InvokerInstrumentation.java (hunk 2 above)
Comment (amarziali) [reply dd-java-agent/instrumentation/cxf-2.1/src/main/java/datadog/trace/instrumentation/cxf/InvokerInstrumentation.java]: done

File: dd-java-agent/instrumentation/cxf-2.1/src/main/java/datadog/trace/instrumentation/cxf/ServletHelper.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
+package datadog.trace.instrumentation.cxf;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// Probably one day we should put it in the bootstrap if used in a lot of places
```
Comment (mcculls) [inline dd-java-agent/instrumentation/cxf-2.1/src/main/java/datadog/trace/instrumentation/cxf/ServletHelper.java:8]: As part of refactoring the build I'd like to make it easier to share utilities based on libraries like the servlet-api. And ideally do this in a way such that the right utility is chosen at advice insertion time, which would reduce the need for method handles.

**Review by mcculls — APPROVED** [review]
The use of method handles here to handle `javax.servet` vs `jakarta.servlet` deployments is acceptable for the short-term. In the long-term I hope to make improvements to the build which would allow us to remove these.

**Pull request merged**